packages/main/src/RPA/Database.py

```diff
@@ -1,23 +1,527 @@
+        charset,
+        config_file: str,
+    ):
+        config = ConfigParser.ConfigParser()
+        config.read([config_file])
+
+        self.configuration = {}
+        self.module_name = module_name or (
+            config.get("default", "module_name")
+            if config.has_option("default", "module_name")
+            else None
+        )
+        self.configuration["database"] = database or (
+            config.get("default", "database")
+            if config.has_option("default", "database")
+            else None
+        )
+        self.configuration["username"] = username or (
+            config.get("default", "username")
+            if config.has_option("default", "username")
+            else None
+        )
+        self.configuration["password"] = password or (
+            config.get("default", "password")
+            if config.has_option("default", "password")
+            else None
+        )
+        self.configuration["host"] = host or (
+            config.get("default", "host")
+            if config.has_option("default", "host")
+            else None
+        )
+        self.configuration["port"] = port or (
+            int(config.get("default", "port"))
+            if config.has_option("default", "host")
+            else None
+        )
+        self.configuration["charset"] = charset or (
+            config.get("default", "charset")
+            if config.has_option("default", "charset")
+            else None
+        )
+        return self.module_name, self.configuration
+
+    def get(self, param, default=None):
+        return (
+            self.configuration[param] if param in self.configuration.keys() else default
+        )
+
+    def set_val(self, param, value):
+        self.configuration[param] = value
+
+    def all_but_empty(self):
+        new_dict = {}
+        for key, value in dict(self.configuration).items():
+            if value is not None:
+                new_dict[key] = value
+        return new_dict
+
+    def set_default_port(self, port):
+        if (
+            "port" not in self.configuration.keys()
+            or self.configuration["port"] is None
+        ):
+            self.configuration["port"] = int(port)
+
+    def get_connection_parameters_as_string(self, conf=None):
+        configuration = conf or self.configuration
+        parameters = ",".join(
+            "{}={}".format(str(k), str(v)) for k, v in configuration.items()
+        )
+        return "Connecting using : %s.connect(%s)" % (self.module_name, parameters)
+
+
+class Database:
+    """Library handling different database operations."""
+
+    ROBOT_LIBRARY_SCOPE = "GLOBAL"
+
+    def __init__(self) -> None:
         self.logger = logging.getLogger(__name__)
-        DatabaseLibrary.__init__(self, *args, **kwargs)
+        self._dbconnection = None
+        self.db_api_module_name = None
+        self.config = Configuration()
+
+    # pylint: disable=R0915
+    def connect_to_database(  # noqa: C901
```

Comment:
I know this keyword is mostly copied from upstream but splitting it up would make a bit more readable/extendable. Not high prio though.